cypress/cypress/integration/history-links.spec.ts

```diff
@@ -0,0 +1,245 @@
+    });
+    cy.intercept("**/questions/?mentor=*&query=*", {
+      fixture: "response_with_markdown.json",
+    });
+    cy.visit("/");
+    // write msgs
+    cy.get("[data-cy=input-field]").type("Question 1");
+    cy.get("[data-cy=input-send]").trigger("mouseover").click();
+    cy.get("[data-cy=topic-2] button").trigger("mouseover").click();
+    cy.get("[data-cy=history-chat]").should("exist");
+    cy.get("[data-cy=chat-msg-2]").contains("Click https://www.google.com");
+    cy.get("[data-cy=chat-msg-2] a").should(
+      "have.attr",
+      "href",
+      "https://www.google.com"
+    );
+    cy.get("[data-cy=chat-msg-2] a").should("have.attr", "target", "_blank");
+  });
+
+  it("Display link-label over a corner of the video", () => {
+    mockDefaultSetup(cy, {
+      config: { mentorsDefault: ["clint", "carlos"] },
+      mentorData: [clint, carlos],
+      apiResponse: "response_with_markdown2.json",
+      gqlQueries: [cyMockGQL("userQuestionSetFeedback", null)],
+    });
+    cy.visit("/");
+    cy.intercept("**/questions/?mentor=clint&query=*", {
+      fixture: "response_with_markdown.json",
+    });
+    cy.intercept("**/questions/?mentor=carlos&query=*", {
+      fixture: "response_with_markdown2.json",
+    });
+    cy.get("[data-cy=answer-link-card]").should("not.exist");
+    cy.get("[data-cy=topic-2] button").trigger("mouseover").click();
+    cy.get("[data-cy=history-chat]").should("exist");
+    // write msgs
+    cy.get("[data-cy=input-field]").type("Question 1");
+    cy.get("[data-cy=input-send]").trigger("mouseover").click();
+    // display label over a corner of the video
+    cy.get("[data-cy=answer-link-card]").should("exist");
+  });
+
+  it("It doesn't display link-label over a corner of the video", () => {
+    visitAsGuestWithDefaultSetup(cy, "/");
+    cy.get("[data-cy=answer-link-card]").should("not.exist");
+    cy.get("[data-cy=topic-2] button").trigger("mouseover").click();
+    cy.get("[data-cy=history-chat]").should("exist");
+    // write msgs
+    cy.get("[data-cy=input-field]").type("Question 1");
+    cy.get("[data-cy=input-send]").trigger("mouseover").click();
+    // display label over a corner of the video
+    cy.get("[data-cy=answer-link-card]").should("not.exist");
+  });
+
+  it("Display link-label for the duration of the video & ensure clears link when a new video", () => {
+    visitAsGuestWithDefaultSetup(cy, "/");
+    cy.intercept("**/questions/?mentor=clint&query=*", {
+      fixture: "response_with_markdown.json",
+    });
+    cy.get("[data-cy=topic-2] button").trigger("mouseover").click();
+    cy.get("[data-cy=history-chat]").should("exist");
+    // write msgs
+    cy.get("[data-cy=input-field]").type("Question 1");
+    cy.get("[data-cy=input-send]").trigger("mouseover").click();
+    cy.get("[data-cy=video-thumbnail-carlos]").trigger("mouseover").click();
+    cy.get("[data-cy=answer-link-card]", { timeout: 8000 }).should("not.exist");
```

Comment:
do we need this super-long timeout? what is it for?

Reply:
We need to wait for the video to finish and then rectify if the link card was removed. This is the reason I added 8000ms because that's what the video lasts.